lib/ssh_scan/config_loader/yaml.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+
+module SSHScan
+  module ConfigLoader
+    class Yaml
+      def initialize(config, class_name = nil, mode = 'file', logger = nil)
+        @opts = case mode
+          when "file" then YAML.load_file(config)
+          when "string" then YAML.load(config)
+        end
+        @logger = logger || Logger.new(STDOUT)
+        @class_name = class_name ? class_name.name : ""
+        @good_opts = loadGoodOpts
+      end
+
+      def opts
+        @opts
+      end
+
+      def checkedOpts
+        unless @good_opts.nil?
+          allowed = @good_opts["allowed"]
+          error_count = 0
+          retrieved_opts = opts
+          retrieved_opts.each do |key, value|
+            unless allowed.include?(key)
+              @logger.error("attribute not allowed: #{key}")
+              error_count += 1
+            end
+          end
+          @logger.error("error count for #{@class_name} configuration: #{error_count}") if error_count > 0
```

Comment:
Config loading errors should be hard failures, if we cannot load the config, there is no point in proceeding.